stream.go

```diff
@@ -0,0 +1,99 @@
+package simpleblob
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"sync"
+)
+
+// A ReaderStorage is an Interface providing an optimized way to create an io.ReadCloser.
+type ReaderStorage interface {
```

Comment:
Nit: perhaps `StorageReader` and `StorageWriter` are better, to make it end with the verb?

Reply:
Or perhaps `StreamReader` and `StreamWriter`?

Reply:
I tried to follow more or less the way it's done with `io/fs`, e.g. `fs.StatFS` is an `fs.FS` that also implements an optimized `Stat` method, so `ReaderStorage` is a `Storage` that also implements an optimized `NewReader` method.

Reply:
This comment still stands, it should end with a verb. `StreamReader` or `StorageReader` is more clear about what it does than `ReaderStorage`, which suggests that it is some kind of storage of/for readers.